rspace/src/main/scala/coop/rchain/rspace/RSpace.scala

```diff
@@ -52,6 +52,100 @@ class RSpace[F[_], C, P, A, K](
   private[this] val produceCommLabel                      = "comm.produce"
   private[this] val produceTimeCommLabel                  = "comm.produce-time"
 
+  override def consume(
+      channels: Seq[C],
+      patterns: Seq[P],
+      continuation: K,
+      persist: Boolean,
+      peeks: SortedSet[Int] = SortedSet.empty
+  ): F[MaybeActionResult] =
+    contextShift.evalOn(scheduler) {
+      if (channels.isEmpty) {
```

Comment:
Yeah, right - so use a data structure that does not permit emptiness - also in the spirit of "making illegal states unrepresentable", one should introduce a structure that enforces that channels and patterns are of the same length (eg. a non-empty sequence of pairs)